src/agent/inspectordebugapi.ts

```diff
@@ -127,11 +130,27 @@ export class InspectorDebugApi implements debugapi.DebugApi {
           cb, breakpoint, StatusMessage.BREAKPOINT_CONDITION,
           utils.messages.V8_BREAKPOINT_CLEAR_ERROR);
     }
-    const id = breakpoint.id;
-    const result = this.v8Inspector.removeBreakpoint(breakpointData.id);
-    const breakpointId = this.breakpoints[id].id;
-    delete this.breakpoints[id];
-    delete this.listeners[breakpointId];
+    let locationStr = breakpointData.locationStr;
+    const bpId = breakpointData.id;
+
+    // delete breakpoint from locationmapper and breakpointmapper.
+    this.deleteBreakpointInArray(
```

Comment:
It looks like it would be advantageous to have `locationmapper` and `breakpointmapper` map keys to a `Map<string>` instead of a `string[]` to support deletion here.